File: backend/src/main/java/com/app/mevocab/internal/result/entity/Result.java
package com.app.mevocab.internal.result.entity;

import com.app.mevocab.internal.quiz.dto.Question;
import com.app.mevocab.internal.quiz.entity.Quiz;
import com.app.mevocab.internal.result.dto.Answer;
import com.app.mevocab.internal.topic.entity.Topic;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.Map;

@Document
@Builder(toBuilder = true)
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Result {
    @Id
    String resultID;
    Quiz quiz;
    String studentID;
    Map<Integer, Answer> answers;
    int totalQuestion;
    int totalCorrect;
    double score;
    Instant createAt;
    Instant finish;
}

File: frontend/lib/data/models/result.dart
import 'package:frontend/data/models/quiz.dart';

class Result {
  final String resultID;
  final Quiz quiz;
  final String studentID;
  final List<Answer> answers;
  final int totalQuestion;
  final int totalCorrect;
  final double score;
  final DateTime createdAt;
  final DateTime finish;

  Result({
    required this.resultID,
    required this.quiz,
    required this.studentID,
    required this.answers,
    required this.totalQuestion,
    required this.totalCorrect,
    required this.score,
    required this.createdAt,
    required this.finish,
  });
}

class Answer {
  final String answerID;
  final String answer;
  final bool isCorrect;

  Answer({
    required this.answerID,
    required this.answer,
    required this.isCorrect,
  });
}
